### DISCUSSION TITLE
ollama server how to run on startup

### DISCUSSION THREAD

**Discussion by DiamondSamurai0** [logancyang/obsidian-copilot#419] [Q&A] [answered]
ive realised  that i have to keep inputting this command everytime i want to use it has anyone found a way to make it run on bootup so the server is up indefinitely or some way to manage it?

**Comment by Nonetss** [comment]
Hi,

I found a solution that might help you with running the Ollama server on startup. I came across this solution while trying to solve another issue I was facing, but I believe it should also work for your case.

You need to modify the `/etc/systemd/system/ollama.service` file. In the `[Service]` section, add the following two lines:

```
ExecStart=/usr/local/bin/ollama serve
Environment="OLLAMA_ORIGINS=app://obsidian.md*"
```

This configuration should allow the Ollama server to start automatically at bootup.

**Reply by DiamondSamurai0** [reply]
This is Linux right I've done similar stuff before but in currently on windows if I find a way I'll post all the instructions

**Reply by Pathsis** [reply]
@Nonetss Thank you very much! I'm running ollama on Ubuntu, using your method, plus running

```shell 
systemctl daemon-reload 
systemctl restart ollama
``` 
I successfully used copilot on Obsidian!

**Comment by DiamondSamurai0** [comment, marked as answer]
https://www.howto-outlook.com/howto/powershell-scripts-faq-tips-and-tricks.htm#google_vignette

is a tuturial to add tasks to windows and set the command in the file to:
$env:OLLAMA_ORIGINS="app://obsidian.md*"; ollama serve

**Comment by joshmedeski** [comment]
The macOS Ollama application is managed by `launchctl`. Update the env for `OLLAMA_ORIGINS` with the following command:

```sh
launchctl setenv OLLAMA_ORIGINS "app://obsidian.md*"
```

Then, restart the Ollama application.
